### DISCUSSION TITLE
Is it possible to enable repeating keys only for the specific IDE where the plugin is installed?

### DISCUSSION THREAD

**Discussion by cartesian-plane** [JetBrains/ideavim#1050] [Q&A] [answered]
Upon first installing the plugin in PyCharm(2024.2.4) on macOS 15.01, a [pop-up](https://github.com/JetBrains/ideavim/blob/03996dce59010416a1c9ea000cd917623492d89a/src/main/java/com/maddyhome/idea/vim/group/NotificationService.kt#L135) appears asking about enabling repeating keys
on press and hold.
Clicking `Yes` changes the global default value of the `ApplePressAndHoldEnabled` to `0`, which seems unnecessary, as there should also be an option to change it on a per-app basis.

I have tried looking for an IDE-specific default, like `com.jetbrains.pycharm ApplePressAndHoldEnabled`, but I could not find it using `defaults`.
Is anyone aware of a better solution for this?

**Comment by citizenmatt** [comment, marked as answer]
Yes, you can set it per-application. The tricky part is finding the application ID. [This gist](https://gist.github.com/lsd/1e1826907ab7e49c536a) has some details. The EAP builds usually have `-EAP` appended, such as `com.jetbrains.rider-EAP`. And you can use a command line such as `defaults read | egrep -o 'com\.jetbrains\.[^\"]+' | sort --unique` to print out all the known IDs for the different IDEs, so you can then set it per-IDE.

Ideally, the plugin would do this itself, but it's unclear how to get the app ID programmatically. An even better solution would be for the plugin to disable press and hold while in Normal mode. I think this is now supported by the JBR, so it should be possible to enable/disable input methods for the editor when the mode changes. However, that needs implementing.

**Comment by 0bvim** [comment]
This works to me: https://www.notion.so/Disable-repeating-keys-32a5b68e255980d3864acab79f2c2988
